src/System.Linq/tests/AverageTests.cs

```diff
@@ -39,6 +39,15 @@ public void EmptyNullableFloatSource()
         }
 
         [Fact]
+        public void EmptyNullableFloatSourceWithSelector()
```

Comment:
I'm having trouble commenting due to GitHub not displaying the Enumerable.cs diff.  So I've picked a random place to comment :smile: ...

In ```Min<TSource, TResult>```, there's this code:
```C#
            if (value == null)
            {
                using (IEnumerator<TSource> e = source.GetEnumerator())
                {
                    do
                    {
                        if (!e.MoveNext()) return value;
                        value = selector(e.Current);
                    } while (value == null);
                    while (e.MoveNext())
                    {
                        TResult x = selector(e.Current);
                        if (x != null && comparer.Compare(x, value) < 0) value = x;
                    }
                }
            }
            else
            {
                using (IEnumerator<TSource> e = source.GetEnumerator())
                {
                    if (!e.MoveNext()) throw Error.NoElements();
                    value = selector(e.Current);
                    while (e.MoveNext())
                    {
                        TResult x = selector(e.Current);
                        if (comparer.Compare(x, value) < 0) value = x;
                    }
                }
            }
```
Why duplicate the using rather than having the ```if (value == null)``` check inside the using?